samza-core/src/main/java/org/apache/samza/coordinator/stream/CoordinatorStreamValueSerde.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.coordinator.stream;
+
+import com.google.common.base.Preconditions;
+import java.util.Map;
+import org.apache.samza.config.Config;
+import org.apache.samza.coordinator.stream.messages.CoordinatorStreamMessage;
+import org.apache.samza.coordinator.stream.messages.SetChangelogMapping;
+import org.apache.samza.coordinator.stream.messages.SetContainerHostMapping;
+import org.apache.samza.coordinator.stream.messages.SetTaskContainerMapping;
+import org.apache.samza.SamzaException;
+import org.apache.samza.serializers.JsonSerde;
+import org.apache.samza.serializers.Serde;
+
+/**
+ * Serializer for values written into the coordinator stream(kafka topic). CoordinatorStreamMessage combines both key
+ * and value serializer for coordinator stream messages. Since value is relevant to this serializer, coordinator stream
+ * key is nuked for different message types.
+ */
+public class CoordinatorStreamValueSerde implements Serde<String> {
+
+  private static final String SOURCE = "SamzaContainer";
+
+  private final String type;
+  private final Serde<Map<String, Object>> messageSerde;
+  private final Config config;
+
+  public CoordinatorStreamValueSerde(Config config, String type) {
+    Preconditions.checkNotNull(type);
+    this.config = config;
+    this.type = type;
+    messageSerde = new JsonSerde<>();
+  }
+
+  @Override
+  public String fromBytes(byte[] bytes) {
+    Map<String, Object> values = messageSerde.fromBytes(bytes);
+    CoordinatorStreamMessage message = new CoordinatorStreamMessage(new Object[]{}, values);
+    if (type.equalsIgnoreCase(SetContainerHostMapping.TYPE)) {
```

Comment:
This is called for EACH message. I think we should optimize this code performance wise (may be use integer for type).

Reply:
Agree. You're right with respect to the performance. 

But the number of messages in coordinator stream is supposed to small(due to log-compaction guarantees from kafka). 

Existing implementation uses the same comparator for each message. We have not seen any performance problems with that setup yet. Moreover, I don't want to alter the existing comparator behavior as a part of this change. If acceptable, will create a followup JIRA and see if we really want to do this.